### DISCUSSION TITLE
How to add COMMANDLINE_ARGS?

### DISCUSSION THREAD

**Discussion by Holyniwa** [vladmandic/sdnext#134] [Q&A] [answered]
I opened up the bat for this program, but it's so vastly different I'm not sure how its suppose to be edited...
is it possible to use things like --lowvram --autolaunch and other COMMANDLINE_ARGS in automatic?

**Comment by vladmandic** [comment]
why not just set `COMMANDLINE_ARGS` in your system environment, then script will happy parse and use it.
or you could create a another batch, like `my-webui.bat` and have single line like `webui.bat --lowvram` or whatever.

**Comment by Holyniwa** [comment]
I'm not sure what to do with that info since I'm coding illiterate, so i asked chatGPT and it gave me this. It seems to work now. Thanks.



@echo off

if not defined PYTHON (set PYTHON=python)
if not defined VENV_DIR (set "VENV_DIR=%~dp0%venv")

set ERROR_REPORTING=FALSE
set COMMANDLINE_ARGS= --lowvram --autolaunch --precision full --no-half --no-half-vae 

mkdir tmp 2>NUL

%PYTHON% -c "" >tmp/stdout.txt 2>tmp/stderr.txt
if %ERRORLEVEL% == 0 goto :check_pip
echo Couldn't launch python
goto :show_stdout_stderr

...
...
...

:launch
%PYTHON% launch.py %COMMANDLINE_ARGS% %*
pause
exit /b

:accelerate_launch
echo Accelerating
%ACCELERATE% launch --num_cpu_threads_per_process=6 launch.py %COMMANDLINE_ARGS%
pause
exit /b

...
...
...

**Reply by vladmandic** [reply, marked as answer]
that bypassed actual `webui.bat`, not recommended.
like i said, create `my-webui.bat` with a single line

```batch
webui.bat --lowvram --autolaunch --precision full --no-half --no-half-vae
```

**Reply by Holyniwa** [reply]
aaaaaaaaaaah~ i took the long bus there, but now i see what you mean. this works better, thank you.

**Comment by netdzynr** [comment]
Following up on this topic for MacOS (M1).  I get this error when when attempting to generate:

`RuntimeError: Input type (c10::Half) and bias type (float) should be the same`

I'm guessing this is due needing the same commandline args I use in auto1111.  Following the above posted instructions, I assumed I needed to create `my-webui.sh` with a single line of args as described, but continue to get the same error.  I also tried creating `my-webui.bat` as well, but no difference.  Is a different config needed for MacOS?

**Reply by vladmandic** [reply]
first make sure that those args even exist here as many are moved to settings.

**Reply by netdzynr** [reply]
Thank you -- args available in Settings as you explained.
